### DISCUSSION TITLE
Is Mermaid timeline graph supported?

### DISCUSSION THREAD

**Discussion by heitorlessa** [squidfunk/mkdocs-material#5708] [Q&A] [answered]
Hey! I've just tried using the [Timeline](https://mermaid.js.org/syntax/timeline.html) graph and it fails to render in mkdocs-material - Is it perhaps a configuration or unsupported? 

This was [introduced in Mermaid 9.4.0](https://github.com/mermaid-js/mermaid/releases/tag/v9.4.0) so it doesn't seem to warrant an upgrade (mkdocs-material is using 9.4.3).

<img width="696" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/3340292/eb13329e-c387-482d-84b2-cf1bc8655b0a">

> NOTE: [Working example in GitHub](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/MAINTAINERS.md#release-process-visualized).

I couldn't find when this feature was introduced to confirm whether that is simply an upgrade

---

Demo example

```mermaid
timeline
        title England's History Timeline
        section Stone Age
          7600 BC : Britain's oldest known house was built in Orkney, Scotland
          6000 BC : Sea levels rise and Britain becomes an island.<br> The people who live here are hunter-gatherers.
        section Bronze Age
          2300 BC : People arrive from Europe and settle in Britain. <br>They bring farming and metalworking.
                  : New styles of pottery and ways of burying the dead appear.
          2200 BC : The last major building works are completed at Stonehenge.<br> People now bury their dead in stone circles.
                  : The first metal objects are made in Britain.Some other nice things happen. it is a good time to be alive.
```

Thanks a lot!

**Comment by facelessuser** [comment, marked as answer]
IIRC, Timelines require using JS modules which MkDocs does not yet support (but may support soon). Assuming MkDocs gets the appropriate support, Material would have to update its approach to use modules.

So, as I understand, Material does not currently support Timelines, but may in the future.

**Reply by squidfunk** [reply]
Jup, correct. We're waiting for module support, and then we can re-evaluate if/how we can migrate to Mermaid.js 10. I think it'll be coming in MkDocs 1.5.0, but I'm not sure when that's going to be released.

**Comment by heitorlessa** [comment]
Gotcha - thank you so much 🙏
